docs/node-observ-lib/main.libsonnet

```diff
@@ -0,0 +1,160 @@
+local alerts = import './alerts.libsonnet';
+local annotations = import './annotations.libsonnet';
+local dashboards = import './dashboards.libsonnet';
+local datasources = import './datasources.libsonnet';
+local g = import './g.libsonnet';
+local links = import './links.libsonnet';
+local panels = import './panels.libsonnet';
+local rules = import './rules.libsonnet';
+local targets = import './targets.libsonnet';
+local variables = import './variables.libsonnet';
+local commonlib = import 'common-lib/common/main.libsonnet';
+
+{
+  withConfigMixin(config): {
+    config+: config,
+  },
+
+  //  any modular observability library should inlcude as inputs:
+  // 'dashboardNamePrefix' - Use as prefix for all Dashboards and (optional) rule groups
+  // 'filteringSelector' - Static selector to apply to ALL dashboard variables of type query, panel queries, alerts and recording rules.
+  // 'groupLabels' - one or more labels that can be used to identify 'group' of instances. In simple cases, can be 'job' or 'cluster'.
+  // 'instanceLabels' - one or more labels that can be used to identify single entity of instances. In simple cases, can be 'instance' or 'pod'.
+  // 'uid' - UID to prefix all dashboards original uids
+
+  new(
+    filteringSelector='job="node"',
+    groupLabels=['job'],
+    instanceLabels=['instance'],
+    dashboardNamePrefix='Node exporter / ',
+    dashboardTags=[uid],
+    uid,
+  ): {
+
+    local this = self,
+    config: {
```

Comment:
Did we decide and document somewhere that config should go in the main definition of the mixin?

We have an existing pattern of looking for an external `config.libsonnet` file to change configurable options, so this could be confusing.

Reply:
Nope, there was no decision if config should be inline or in separate file. I'll see what I can do.

Reply:
Decided to refactor approach. new() is now parameterless, and all config is moved to config.libsonnet file. Some sane defaults are provided for all params required in modular libs. And any of those defaults can be modified by by calling it like this:
```
local g = import './g.libsonnet';
local nodelib = import './main.libsonnet';


local linux =
  nodelib.new()
  + nodelib.withConfigMixin({
    filteringSelector: 'job=~".*node.*"',
    groupLabels: ['job'],
    instanceLabels: ['instance'],
    dashboardNamePrefix: 'Node exporter / ',
    dashboardTags: ['node-exporter-mixin'],
    uid: 'node',
  });

```

Reply:
This looks great to me! Makes the config a bit more modular and clear where the changeable parameters are.

I like the `withConfigMixin` function too, allowing the user to easily supply their own config overrides.

This makes sense to me, I'd be curious if the rest of the team has any feedback. @Dasomeone @gaantunes ?